py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,424 @@
+
+        Returns: Credentials, the obtained credential.
+        """
+        client_secret_path = os.path.dirname(os.path.expanduser(self.client_secret))
+        auth_token_path = os.path.dirname(os.path.expanduser(self.auth_token))
+
+        if not os.path.exists(auth_token_path):
+            os.makedirs(auth_token_path)
+
+        if not os.path.exists(client_secret_path):
+            os.makedirs(client_secret_path)
+
+        self.auth_token = os.path.expanduser(self.auth_token)
+        self.client_secret = os.path.expanduser(self.client_secret)
+
+        flags = tools.argparser.parse_args(args=[])
+        store = Storage(self.auth_token)
+        credentials = store.get()
+
+        if not credentials or credentials.invalid:
+            flow = client.flow_from_clientsecrets(self.client_secret, SCOPES)
+            flow.user_agent = APPLICATION_NAME
+            if flags:
+                credentials = tools.run_flow(flow, store, flags)
+            else:  # Needed only for compatibility with Python 2.6
+                credentials = tools.run(flow, store)
+
+            """
+            Have to restart i3 after getting credentials to prevent bad output.
+            This only has to be done once on the first run of the module.
+            """
+            self.py3.command_output('i3-msg restart')
+
+        return credentials
+
+    def _authorize_credentials(self):
+        """
+        Fetches an access/refresh token by authorizing OAuth 2.0 credentials.
+
+        Returns: True, if the authorization was successful.
+                 False, if a ServerNotFoundError is thrown.
+        """
+        try:
+            http = self.credentials.authorize(httplib2.Http())
+            self.service = discovery.build('calendar', 'v3', http=http)
+            return True
+        except ServerNotFoundError:
+            return False
+
+    def _get_events(self):
+        """
+        Fetches events from the color into a list.
+
+        Returns: The list of events.
+        """
+        now = datetime.datetime.utcnow().isoformat() + 'Z'  # 'Z' indicates UTC time
+
+        eventsResult = self.service.events().list(
+            calendarId='primary', timeMin=now, maxResults=self.num_events, singleEvents=True,
+            orderBy='startTime').execute()
+        return eventsResult.get('items', [])
+
+    def _check_warn_threshold(self, time_to, summary):
+        """
+        Checks if the time until an event starts is less than or equal to the warn_threshold.
+        If it is, it presents a warning with self.py3.notify_user.
+        """
+        if time_to['total_minutes'] <= self.warn_threshold:
+            self.py3.notify_user("Warning: Appointment " + str(summary) +
+                                 " coming up!", 'warning', self.warn_timeout)
+
+    def _gstr_to_datetime(self, date_time_str):
+        """ Returns a datetime object from a google calendar date/time string."""
+        tmp = '-'.join(date_time_str.split('-')[:-1])
```

Comment:
```
2017-10-10 01:31:09 WARNING Instance `google_calendar`,
        user method `google_calendar` failed (AttributeError) google_calendar.py line 236.
2017-10-10 01:31:09 INFO Traceback
AttributeError: 'NoneType' object has no attribute 'split'
  File "/home/chris/src/py3status/py3status/module.py", line 729, in run
    response = method()
  File ".../modules/google_calendar.py", line 395, in google_calendar
  File ".../modules/google_calendar.py", line 321, in _build_response
  File ".../modules/google_calendar.py", line 236, in _gstr_to_datetime
```

Reply:
Hmm, that is strange. I can't reproduce it on my end. It looks like it's not fetching `dateTime` entries for the event.

Reply:
Right you are. I logged.

We have `'start': {'date': '2017-10-10'},` instead of the expecting `  'start': {'dateTime': '2017-10-10T04:00:00-05:00'},` as this one is an all-day event. Time not needed here.

Reply:
Ah there it is. I never use those so it never came up. Will fix tomorrow.